New shape control with label and popover panel (#897)
makeswift/makeswift: .changeset/hot-pumpkins-clean.md

---
'@makeswift/controls': minor
'@makeswift/runtime': minor
---

## Breaking change

### Updated `Shape` control

The `Shape` control has been updated with two new params `label` and `layout`.

- `label?: string = "Group"`
  - Text for the panel label in the Makeswift builder.

>   - This `label` defaults to "Group."

Reviewer: ```suggestion
  - This `label` defaults to `"Group"`.
```

> - `layout?: 'Shape.Layout.Inline | Shape.Layout.Popover = Shape.Layout.Popover`

Reviewer: ```suggestion
- `layout?: Shape.Layout.Inline | Shape.Layout.Popover = Shape.Layout.Popover`
```

>   - The configuration for how panels are rendered in the right sidebar.

Reviewer: ```suggestion
  - The configuration for how panels are rendered in the Properties Sidebar.
```

We should use the same naming we use in our docs

Author: 💯. I agree.  I couldn't find [this page](https://docs.makeswift.com/product/builder-basics#the-basics) at the time, when I was searching there were quite a few "right sidebar" usages, but I like this change.

![CleanShot 2024-12-20 at 14 45 32@2x](https://github.com/user-attachments/assets/95c00a42-7268-4268-b158-06800cff3662)

Author: Ping @jamesqquick. It might be good to update these "right sidebar" instances to say Properties sidebar.

  - The `Inline` option is the default and it is similar to the current rendering style of the `Shape` popover.
  - The `Popover` option is a new rendering style that only shows your label with a button in the current context. When this button is clicked a popover opens to the left revealing the controls within your `Shape`.

### How to migrate

> This change is listed as minor version because it visually changes how your panels are rendered and we no longer support the old version. Upgrading the `@makeswift/runtime` to this version will update how the right sidebar looks for a `Shape` control to include a `label` and have better visual hierarchy.

Reviewer: ```suggestion
This change is listed as a breaking change because it visually changes how your panels are rendered and we no longer support the old version. Upgrading `@makeswift/runtime` to this version will update how the Properties Sidebar looks for a `Shape` control to include a `label` and have a better visual hierarchy.
```

> This breaking change doesn't require any code changes, but we do suggest adding descriptive labels since the `Shape` control's `label` will default to "Group."

Reviewer: ```suggestion
This breaking change doesn't require any code changes, but we do suggest adding descriptive labels since the `Shape` control's `label` will default to `"Group"`.
```

Reviewer: Same with this one, if we're indicating a default value that's a string, wouldn't we want to put it as "code"? Also, I understand that the period inside quotes is grammatically the correct thing to do, but I think this would be confusing for readers and it's another reason to do '`"Group"`.'

Anyway, this is a nit, just trying to keep things clear for users reading this.

I just wasn't sure if you meant to address this given you didn't respond
